### ISSUE TITLE
Create validation set (self-play and professional)

### ISSUE THREAD

**Issue by brilee** [tensorflow/minigo#43]
For some small percent of self-play games, we should set them aside, never train on them, and check the value net loss on these games. This should help us figure out if we're overfitting.

A variant on this is to create a validation set of pro games. Self-play validation sets are nice because there's not a very large library of pro 9x9 games. But for 19x19, a pro validation set is doable.

**Comment by brilee** [comment]
Reopening; we can mark fixed when we've actually got another process automatically processing tfrecord file, implemented a `get_eval_input` function in preprocessing.py, and hook up eval/train to alternate running in dual_net.train().

**Comment by amj** [comment]
Sounds good.  Should we do this at the end of a training run, basically?  as a separate process?

**Comment by brilee** [comment]
Yeah, it could be as simple as train() then eval(). We don't need that much
granularity; one data point per generation is good enough.

**Comment by amj** [comment]
It'd be great to also get a datapoint on how the rotation/reflection bias is trending.  Could be as simple as take a position, do all 8 rotations, compute the stddev of the 8 value estimates, and do that as part as eval()

**Comment by amj** [comment]
(The 'rotation bias' issue is in #84 )
